### DISCUSSION TITLE
Clear step by step instructions on building a certain version of Ctrlr in Visual Studio

### DISCUSSION THREAD

**Discussion by dobo365** [RomanKubiak/ctrlr#97] [Q&A]
Hi,
  I would gladly support the development/improvement of Ctrlr with simple things.
I'm using 5.3.201 and at this stage do not plan to use the latest version (of course this may change).
I'm not familiar enough with the usage of different versions and Visual Studio and thus would it be possible to get **clear step by step instructions on building a certain version of Ctrlr in Visual Studio?**
Of course focusing on 5.3.201 first then the last version.
Thx in advance!
(I have a similar but different request that I post in another discussion)

**Comment by shooking** [comment]
This would be great.
For example I pulled ctrlr to Linux box today. It doesnt build.
When I look in the Ctrlr.jucer I can see it wants to find

          <FILE id="EVBBxM" name="search.svg" compile="0" resource="1" file="Source/Resources/Icons/search.svg"/>
          <FILE id="dai4Dv" name="search_off.svg" compile="0" resource="1" file="Source/Resources/Icons/search_off.svg"/>
          <FILE id="ky2beP" name="send.svg" compile="0" resource="1" file="Source/Resources/Icons/send.svg"/>

but these files dont exist in the source. So clearly it cannot build. I dont have any problem about a dev branch not being buildable.

What tips on finding a working release or suitable git checkout point does anyone recommend please?

**Comment by shooking** [comment]
I was able to build RPi clean today once I regenerated Jucer.
So I am testing on Linux and Windows.
On Windows I pull down JUCE git, build it.
Mine is installed in

`c:\Pilots\JUCE\extras\Projucer\Builds\VisualStudio2019\x64\Debug\App>.\Projucer.exe`

So now I can regenerate the Ctrlr.jucer (I tried without regen and build fails - cannot find the modules).

```
c:\Pilots\ctrlr\Builds\VisualStudio2019
msbuild Ctrlr.sln /t:Rebuild
...
...

"c:\Pilots\ctrlr\Builds\VisualStudio2019\Ctrlr.sln" (Rebuild target) (1) ->
"c:\Pilots\ctrlr\Builds\VisualStudio2019\Ctrlr_StandalonePlugin.vcxproj.metaproj" (Rebuild target) (2) ->
"c:\Pilots\ctrlr\Builds\VisualStudio2019\Ctrlr_SharedCode.vcxproj" (Rebuild target) (3) ->
(ClCompile target) ->
  c:\Pilots\ctrlr\Source\Native\CtrlrWindows.cpp(93,35): error C2039: 'ico_midi_png': is not a member of 'BinaryData' [c:\Pilots\ctrlr\Builds\Visua
lStudio2019\Ctrlr_SharedCode.vcxproj]
  c:\Pilots\ctrlr\Source\Native\CtrlrWindows.cpp(93,47): error C2065: 'ico_midi_png': undeclared identifier [c:\Pilots\ctrlr\Builds\VisualStudio201
9\Ctrlr_SharedCode.vcxproj]
  c:\Pilots\ctrlr\Source\Native\CtrlrWindows.cpp(93,61): error C2039: 'ico_midi_pngSize': is not a member of 'BinaryData' [c:\Pilots\ctrlr\Builds\V
isualStudio2019\Ctrlr_SharedCode.vcxproj]
  c:\Pilots\ctrlr\Source\Native\CtrlrWindows.cpp(93,77): error C2065: 'ico_midi_pngSize': undeclared identifier [c:\Pilots\ctrlr\Builds\VisualStudi
o2019\Ctrlr_SharedCode.vcxproj]

    63 Warning(s)
    4 Error(s)

```

Any ideas on what this means?
